### DISCUSSION TITLE
MeshCentral image cache

### DISCUSSION THREAD

**Discussion by DaanSelen** [Ylianst/MeshCentral#6347] [General]
Hello all, I wanted to ask about the image cache MeshCentral uses. I see that when I alter the images in MeshCentral config.json it does not immediatly change, is it loaded somewhere into a cache? And how to clear that cache so I immediatly see it?

**Comment by si458** [comment]
if you change anything inside `config.json` you MUST restart your meshcentral server/service

**Reply by DaanSelen** [reply]
Yes I do this, but as an example, right now I have set an image which I changed (made darker) and made from `jpg` to `png`. However opening the browser in private and what measures I take, I still see the old image, specifically on Welcomepicture field.

**Reply by si458** [reply]
where did you place the `Welcomepicture` image?
as it should be placed inside the `meshcentral-data` folder
https://ylianst.github.io/MeshCentral/meshcentral/customization/#web-branding

**Reply by DaanSelen** [reply]
Yes I did 
```
"welcomePicture": "systemec_datacenter_background.png",
"welcomePictureFullScreen": true,
```
This is a part of the image:
![image](https://github.com/user-attachments/assets/c4339b5d-864c-40e8-a886-7d91588cb7df)

This is the login screen:
![image](https://github.com/user-attachments/assets/2b8ac149-35d2-4443-830d-ceababdcfb12)

The one on it right now was the previous image, which I took back. Darkened and reuploaded.

**Reply by DaanSelen** [reply]
I changed nothing and it suddenly now changed.

**Reply by si458** [reply]
just checked the source code and we dont do any caching as its commented out,
we just hand the browser whatever the file is, 
its possible the browser is caching the file, as the web page just loads `welcome.png`
and then we actually hand back whatever file you set in config.json which can be jpg if you wanted it too
